GVFS/GVFS.Virtualization/Projection/GitIndexProjection.LazyUTF8String.cs

```diff
@@ -138,6 +138,11 @@ public unsafe int CaseInsensitiveCompare(LazyUTF8String other)
                         byte thisC = *thisPtr;
                         byte otherC = *otherPtr;
 
+                        if (caseSensitive)

```

Comment:
@chrisd8088 has the impact of this change on index parsing performance been measured?  From what I remember this path is hot enough that Saeed was able to get measurable perf improvements with every conditional check he was able to remove.

`GVFS.PerfProfiling` can be used to isolate/measure the performance of that code path, and we should verify this code path isn't impacted by the change.  @chrisd8088 let me know if you have a Windows box to test this, if not I can do a few runs for you with and without this change.

Reply:
Thanks for the attentive review, @wilbaker -- performance is a great question!  I think I can just hoist this check right out of the loop, now that I look more closely.  Let me do that, and see what I can do about the SHA casing as well, and then ping you for a Windows performance test run!  Thanks!

Reply:
@chrisd8088 I didn't see a different in the time to rebuild the projection with the latest changes and `GVFS.PerfProfiling`.  Thanks!